app/queries/sortered_products.rb

```diff
@@ -0,0 +1,53 @@
+class SorteredProducts < Rectify::Query
+  include PaginationHelper
+
+  def initialize(params)
+    @params = params
+    @params[:sort] = 'newest' unless @params[:sort].present?
+    @page = @params[:page].present? ? @params[:page].to_i : 1
+    @limit = 12
+    @products = Product.in_stock
+  end
+
+  def query
+    all.limit(@limit).offset(offset).decorate
+  end
+
+  def total_pages
+    total = all.to_a.size
+    (total.to_f / @limit).ceil
```

Comment:
1. no need to use local variable here, just all.to_a.size
2. not clear why do you use to_f and ceil here